gandi/resource_domain.go

```diff
@@ -35,20 +35,24 @@ func resourceDomain() *schema.Resource {
 				Optional:    true,
 				Description: "Should the domain autorenew",
 			},
-			"owner":   contactSchema(true),
-			"admin":   contactSchema(false),
-			"billing": contactSchema(false),
-			"tech":    contactSchema(false),
+			"owner":   contactSchema(false),
+			"admin":   contactSchema(true),
+			"billing": contactSchema(true),
+			"tech":    contactSchema(true),
 		},
 		Timeouts: &schema.ResourceTimeout{Default: schema.DefaultTimeout(1 * time.Minute)},
 	}
 }
 
-func contactSchema(required bool) *schema.Schema {
+func contactSchema(optional bool) *schema.Schema {
 	return &schema.Schema{
 		Type:     schema.TypeSet,
-		Required: required,
-		Optional: !required,
+		Required: !optional,
+		Optional: optional,
+		// If not specified, some (billing, admin, tech)
+		// contacts are generated by the API (it copies the
+		// owner contact information)
```

Comment:
This is by the API itself? Because when I've tried it has complained about missing `admin`, `tech`, and `billing`? Or is this something you are (planning on) doing in the terraform module?

Reply:
Yes, the API is using the owner contact information when billing, tech or admin contact are not provided.

The PR #124 proves this API behavior since it creates a domain without specifying these attributes.

Maybe you had this error during the update, but this is fixed by this PR (https://github.com/go-gandi/terraform-provider-gandi/pull/123/files#diff-984996b5406d59e1d7447de9d9c6f4fe1086dc154ca115fdb85fc46f604ad1aaR55)